garak/generators/ollama.py

```diff
@@ -0,0 +1,83 @@
+"""Ollama interface"""
+
+from typing import List, Union
+
+import backoff
+import ollama
+
+from garak import _config
+from garak.generators.base import Generator
+
+
+def _give_up(error):
+    return isinstance(error, ollama.ResponseError) and error.status_code == 404
+
+
+class OllamaGenerator(Generator):
+    """Interface for Ollama endpoints
+
+    Model names can be passed in short form like "llama2" or specific versions or sizes like "gemma:7b" or "llama2:latest"
+    """
+
+    DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
+        "timeout": 30,  # Add a timeout of 30 seconds. Ollama can tend to hang forever on failures, if this is not present
+        "host": "127.0.0.1:11434",  # The default host of an Ollama server. This should maybe be loaded from a config file somewhere
```

Comment:
If I choose the gen_config.json, is this something I should push to the repo in the config folder?
Or should the default still come from the DEFAULT_PARAMS dict, in case there is no config file provided?

Reply:
There is no need to add anything to the config folder for this PR. The example configs were just offered to explain it is already possible to provide a config file on the command line that will allow the user to inject a `host` value. This class should definitely retain a default value.

Reply:
Alright. With regards to the format, IP:PORT works but also fx localhost:{PORT}
I guess I could set up some IP and PORT validation, and then just include "localhost" as a valid value as well?

Reply:
Looking at the client [config examples from ollama](https://github.com/ollama/ollama-python/blob/1ec88ed99466d6c232445714e4eabe99db0e166c/ollama/_client.py#L966-L1027) I do not believe a `port` is required, I can see deferring this for now leaving the default and possibly restricting in a future iteration if the project starts getting feedback that users are getting errors they cannot debug when passing values that cannot be resolved by the client.

Reply:
Alright I left the default as is. Does this mean I can resolve this comment?